crawl-ref/source/view.cc

```diff
@@ -1649,6 +1650,19 @@ static void _config_layers_menu()
                       _layers_saved = _layers |= LAYER_MONSTERS;
                   break;
 #endif
+        CASE_ESCAPE if (_layers)
+                      _layers_saved = _layers = LAYERS_NONE;
+                  else
+                  {
+#ifndef USE_TILE_LOCAL
```

Comment:
I preferred some repetition to having the macro modify the inside of the statement; let me know if the latter is considered better.